src/Monomer/Widgets/Containers/Grid.hs

```diff
@@ -105,31 +115,37 @@ makeFixedGrid isHorizontal config = widget where
 
   getDimSizeReq mainAxis accesor vchildren
     | Seq.null vreqs = fixedSize 0
-    | mainAxis = foldl1 sizeReqMergeSum (Seq.replicate nreqs maxSize)
+    | mainAxis = foldl1 sizeReqMergeSum (Seq.replicate nreqs maxSize) & L.fixed %~ (+ totalSpacing)
     | otherwise = maxSize
     where
       vreqs = accesor <$> vchildren
       nreqs = Seq.length vreqs
       ~maxSize = foldl1 sizeReqMergeMax vreqs
+      totalSpacing = fromIntegral (nreqs - 1) * childSpacing
 
   resize wenv node viewport children = resized where
     style = currentStyle wenv node
     contentArea = fromMaybe def (removeOuterBounds style viewport)
     Rect l t w h = contentArea
     vchildren = Seq.filter (_wniVisible . _wnInfo) children
 
+    totalSpacingW = fromIntegral (max 0 (cols - 1)) * childSpacing
+    totalSpacingH = fromIntegral (max 0 (rows - 1)) * childSpacing
+
     cols = if isHorizontal then length vchildren else 1
     rows = if isHorizontal then 1 else length vchildren
 
-    cw = if cols > 0 then w / fromIntegral cols else 0
-    ch = if rows > 0 then h / fromIntegral rows else 0
+    cw = if cols > 0 then (w - totalSpacingW) / fromIntegral cols else 0
+    ch = if rows > 0 then (h - totalSpacingH) / fromIntegral rows else 0
 
     cx i
-      | rows > 0 = l + fromIntegral (i `div` rows) * cw
+      | rows > 0 = l + fromIntegral (i `div` rows) * cw + spacingOffset i
```

Comment:
I think this section should be something like:

```haskell
    cx i
      | rows == 0 = 0
      | isHorizontal = l + fromIntegral (i `div` rows) * cw + spacingOffset i
      | otherwise = l
    cy i
      | cols == 0 = 0
      | isVertical = t + fromIntegral (i `div` cols) * ch + spacingOffset i
      | otherwise = t
```

Otherwise, the secondary axis is affected by the spacing too. You can check it in `Tutorial02_Styling`, which uses a `grid`, by adding `childSpacing` in line 46.

I'm pretty sure the original code, before your proposed changes, worked just by sheer luck (the `div` fixed the logic mistake, I guess).

Reply:
Ah, you are right - this is a bug! I have just pushed your suggestion as a fix (and changed the tests to catch this case).

I am still somewhat confused by the `div`s -- are these not redundant? E.g. when `rows /= 0 && isHorizontal` then `rows` will always be 1 (right?) so ``i `div` rows`` will always be `i`?

Reply:
You're absolutely right, yes! The `div` can be removed.

Reply:
Done :)